lms/static/scripts/frontend_apps/components/ValidationMessage.js

```diff
@@ -0,0 +1,49 @@
+import { createElement } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+import classNames from 'classnames';
+import propTypes from 'prop-types';
+
+/**
+ * Shows a single validation error message that can be open or closed.
+ * A user can also close the message by clicking on it.
+ */
+
+export default function ValidationMessage({
+  message,
+  open,
+  onClose = () => {},
+}) {
+  const [showError, setShowError] = useState(false);
+
+  useEffect(() => {
+    setShowError(open);
+  }, [open]);
+
+  /**
+   * Closes the validation error message and notifies parent
+   */
+  const closeValidationError = () => {
+    setShowError(false);
+    onClose();
+  };
+
+  const errorClass = classNames('ValidationMessage', {
+    'ValidationMessage--open': showError,
+    'ValidationMessage--closed': !showError,
+  });
+
+  return (
+    <div onClick={closeValidationError} className={errorClass}>
+      {message}
+    </div>
+  );
+}
+
+ValidationMessage.propTypes = {
+  // Error message text
+  message: propTypes.string.isRequired,
+  // Should this be open or closed
+  open: propTypes.bool.isRequired,
```

Comment:
Minor note: perhaps `open` could take a default value instead of being required.